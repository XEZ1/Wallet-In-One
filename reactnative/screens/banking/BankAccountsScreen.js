import { StatusBar } from 'expo-status-bar';
import { StyleSheet, Text, View, ScrollView, Button, TextInput, Alert, FlatList , TouchableOpacity, Image} from 'react-native';
import { useState, useEffect } from 'react';
import { auth_get} from '../../authentication'
import { useIsFocused } from '@react-navigation/native';
import Loading from './Loading'
import { useTheme } from 'reactnative/src/theme/ThemeProvider'


export default function BankAccountsScreen({ navigation }) {
  const [ isLoading, setIsLoading ] = useState(true)
  const [ bankData, setBankData ] = useState([])
  const isFocused = useIsFocused();
  const {dark, colors, setScheme} = useTheme();
  
  useEffect(() =>{
    const fetchData = async () => {
        console.log('fetch bank data')
        const response = await auth_get('/banking/user_accounts')
        console.log('response', response)
        if (response.status == 200){
            setIsLoading(false)
            setBankData(response.body)
        }
    }
    fetchData()
  }, [isFocused])

  const displayTimestamp = (timestamp) => {
    const date = new Date(timestamp);
    return  date.toLocaleDateString() + " "+  date.toLocaleTimeString();
  };

  const styles = StyleSheet.create({
    container: {
      width: '100%',
      paddingLeft: 20,
      paddingRight: 20,
      paddingBottom: 0,
      paddingTop: 10,
      //borderWidth: 1,
      borderRadius: 5,
      borderColor: dark ? colors.background : '#ddd',
      overflow: 'hidden',
    },
    row:{
      flexDirection: 'row',
      alignItems: 'flex-start',
    },
    item:{
      padding: 20,
      borderRadius: 10,
    },
    text:{
      color: colors.text
    },
    name:{
      color: "white",
      fontWeight: 'bold',
      fontSize: 21,
    },
    amount:{
      color: "white",
      fontSize: 18,
    },
    iban:{
      color: 'rgba(255,255,255,0.5)',
      fontSize: 10,
      marginBottom: 3,
    },
  });
  
  

  if (isLoading){
    return <Loading/>
}

  return (
    // <View style={{flex:1,  margin: 4, marginBottom: 54, backgroundColor: colors.background}} >
    <View style={{flex:1, backgroundColor: colors.background}} >
              <View style={[styles.container]}>
                  <FlatList data={bankData} renderItem={({item, index}) =>{
                      return (
                        <TouchableOpacity style={[styles.item, {backgroundColor: item.color}]} key={index}>
                            <View style={styles.row}>
                                <Image
                                    source={{ uri: item.institution_logo }}
                                    style={{ width: 70, height: 70, marginRight: 10, resizeMode: 'contain', borderRadius: 10}}
                                />
                                <View style={{ borderRadius: 10}}>
                                  <Text style={styles.name}>{item.institution_name}</Text>
                                  <Text style={styles.iban}>{item.iban}</Text>
                                  <Text style={styles.amount}>{item.balance.string}</Text>
                                </View>
                              </View>
                          </TouchableOpacity>)
                      }}
                      ListEmptyComponent={<Text style={styles.text}>{'\nYou have no bank accounts\n'}</Text>}
                  />
              </View>
      </View>
  );
}