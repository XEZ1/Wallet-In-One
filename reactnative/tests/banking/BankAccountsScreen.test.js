import React from 'react';
import { render, screen, fireEvent, act, within, waitFor} from '@testing-library/react-native';
import BankAccountsScreen from '../../screens/banking/BankAccountsScreen'


describe('<BankAccountsScreen />', () => {
    it('snapshot test', async () => {

        var response = [
            {
                "id": "1",
                "requisition_id": "None",
                "last_update": "2023-03-17T13:32:01.539829Z",
                "iban": "1234",
                "institution_id": "REVOLUT_REVOGB21",
                "institution_name": "Revolut",
                "institution_logo": "https://cdn.nordigen.com/ais/REVOLUT_REVOGB21.png",
                "color": "#18A7F7",
                "disabled": false,
                "user": 1,
                "balance": {
                    "string": "£100",
                    "currency": "GBP",
                    "amount": "100"
                }
            },
            {
                "id": "2",
                "requisition_id": "None",
                "last_update": "2023-03-17T13:32:01.539829Z",
                "iban": "1234",
                "institution_id": "REVOLUT_REVOGB21",
                "institution_name": "Revolut",
                "institution_logo": "https://cdn.nordigen.com/ais/REVOLUT_REVOGB21.png",
                "color": "#18A7F7",
                "disabled": true,
                "user": 1,
                "balance": {
                    "string": "£200",
                    "currency": "GBP",
                    "amount": "200"
                }
            },
        ]
        
        snapshot = render(<BankAccountsScreen testData={response} />);
        expect(snapshot).toMatchSnapshot();
    }
    )
});