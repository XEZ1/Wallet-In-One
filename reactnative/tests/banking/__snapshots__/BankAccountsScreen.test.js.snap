// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BankAccountsScreen /> snapshot test 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "purple",
          "borderRadius": 10,
          "marginHorizontal": 8,
          "marginVertical": 4,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "black",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Bank Insights
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "purple",
          "borderRadius": 10,
          "marginHorizontal": 8,
          "marginVertical": 4,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "black",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        All Transactions
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "borderColor": "#ddd",
          "borderRadius": 5,
          "overflow": "hidden",
          "paddingBottom": 0,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 10,
          "width": "100%",
        },
      ]
    }
  >
    <RCTScrollView
      ListEmptyComponent={
        <Text
          style={
            {
              "color": "black",
            }
          }
        >
          
You have no bank accounts

        </Text>
      }
      data={
        [
          {
            "balance": {
              "amount": "100",
              "currency": "GBP",
              "string": "£100",
            },
            "color": "#18A7F7",
            "disabled": false,
            "iban": "1234",
            "id": "1",
            "institution_id": "REVOLUT_REVOGB21",
            "institution_logo": "https://cdn.nordigen.com/ais/REVOLUT_REVOGB21.png",
            "institution_name": "Revolut",
            "last_update": "2023-03-17T13:32:01.539829Z",
            "requisition_id": "None",
            "user": 1,
          },
          {
            "balance": {
              "amount": "200",
              "currency": "GBP",
              "string": "£200",
            },
            "color": "#18A7F7",
            "disabled": true,
            "iban": "1234",
            "id": "2",
            "institution_id": "REVOLUT_REVOGB21",
            "institution_logo": "https://cdn.nordigen.com/ais/REVOLUT_REVOGB21.png",
            "institution_name": "Revolut",
            "last_update": "2023-03-17T13:32:01.539829Z",
            "requisition_id": "None",
            "user": 1,
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#18A7F7",
                "borderRadius": 10,
                "marginBottom": 10,
                "opacity": 1,
                "padding": 20,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                }
              }
            >
              <Image
                source={
                  {
                    "uri": "https://cdn.nordigen.com/ais/REVOLUT_REVOGB21.png",
                  }
                }
                style={
                  {
                    "backgroundColor": "white",
                    "borderRadius": 10,
                    "height": 70,
                    "marginRight": 10,
                    "resizeMode": "contain",
                    "width": 70,
                  }
                }
              />
              <View
                style={
                  {
                    "borderRadius": 10,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "white",
                      "fontSize": 21,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Revolut
                </Text>
                <Text
                  style={
                    {
                      "color": "rgba(255,255,255,0.5)",
                      "fontSize": 10,
                      "marginBottom": 3,
                    }
                  }
                >
                  1234
                </Text>
                <Text
                  style={
                    {
                      "color": "white",
                      "fontSize": 18,
                    }
                  }
                >
                  £100
                </Text>
              </View>
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "red",
                "borderRadius": 15,
                "height": 27,
                "justifyContent": "center",
                "opacity": 1,
                "position": "absolute",
                "right": 10,
                "top": 10,
                "width": 27,
              }
            }
          >
            <Text />
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Modal
              animationType="slide"
              hardwareAccelerated={false}
              onRequestClose={[Function]}
              transparent={true}
              visible={false}
            />
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#18A7F7",
                "borderColor": "red",
                "borderRadius": 10,
                "borderWidth": 4,
                "marginBottom": 10,
                "opacity": 1,
                "padding": 20,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                }
              }
            >
              <Image
                source={
                  {
                    "uri": "https://cdn.nordigen.com/ais/REVOLUT_REVOGB21.png",
                  }
                }
                style={
                  {
                    "backgroundColor": "white",
                    "borderRadius": 10,
                    "height": 70,
                    "marginRight": 10,
                    "resizeMode": "contain",
                    "width": 70,
                  }
                }
              />
              <View
                style={
                  {
                    "borderRadius": 10,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "white",
                      "fontSize": 21,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Revolut
                </Text>
                <Text
                  style={
                    {
                      "color": "rgba(255,255,255,0.5)",
                      "fontSize": 10,
                      "marginBottom": 3,
                    }
                  }
                >
                  1234
                </Text>
                <Text
                  style={
                    {
                      "color": "white",
                      "fontSize": 18,
                    }
                  }
                >
                  £200
                </Text>
              </View>
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "red",
                "borderRadius": 15,
                "height": 27,
                "justifyContent": "center",
                "opacity": 1,
                "position": "absolute",
                "right": 10,
                "top": 10,
                "width": 27,
              }
            }
          >
            <Text />
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Modal
              animationType="slide"
              hardwareAccelerated={false}
              onRequestClose={[Function]}
              transparent={true}
              visible={false}
            />
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#ecba1d",
                "borderRadius": 15,
                "bottom": 20,
                "height": 27,
                "justifyContent": "center",
                "opacity": 1,
                "position": "absolute",
                "right": 10,
                "width": 27,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
